test/ArrayFire.UnitTest/ArrayTest.cs

```diff
@@ -14,5 +14,23 @@ public void Creation()
             var A = Data.Range<int>(2, 2);
             Util.Print(A, "A");
         }
+
+        [TestMethod]
+        public void Dot2x2()
+        {
+            var x = Data.CreateArray(new float[,]
+            {
+                { 1, 0 },
+                { 0, 1 },
+            });
+
+            var y = Data.CreateArray(new float[,]
+            {
+                { 4, 1 },
+                { 2, 2 },
+            });
+
+            var dot = Matrix.MatMul(x, y);
```

Comment:
Perhaps add more line where dot content is compared to expected result also.

Reply:
This PR is just for add Matrix.MatMul interface, not include unit test.